plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1787,6 +1809,25 @@ private void checkUnsupportedGeneratedColumns(MetadataEntry metadataEntry)
         }
     }
 
+    private void checkUnsupportedColumnMappingId(MetadataEntry metadataEntry)
+    {
+        ColumnMappingMode columnMappingMode = getColumnMappingMode(metadataEntry);
+        if (columnMappingMode == ColumnMappingMode.ID) {
+            throw new TrinoException(NOT_SUPPORTED, "Writing with column mapping id is not supported");
+        }
+    }
+
+    private void checkSupportedDmlWriterVersion(ConnectorSession session, DeltaLakeTableHandle table)
+    {
+        SchemaTableName schemaTableName = table.getSchemaTableName();
+        int requiredWriterVersion = getProtocolEntry(session, schemaTableName).getMinWriterVersion();
+        ColumnMappingMode columnMappingMode = getColumnMappingMode(table.getMetadataEntry());
+        if (requiredWriterVersion == MAX_DML_WRITER_VERSION && columnMappingMode != ColumnMappingMode.UNKNOWN) {
```

Comment:
we need to check here that the `columnMappingMode` is specifically `ColumnMappingMode.NAME` here.

Reply:
done